lib/src/zap/sock/zap_sock.c

```diff
@@ -2030,8 +2030,8 @@ zap_err_t zap_transport_get(zap_t *pz, zap_log_fn_t log_fn,
 	if (!z)
 		goto err;
 
-	/* max_msg is unused (since RDMA) ... */
-	z->max_msg = (1024 * 1024) - sizeof(struct sock_msg_hdr);
+	/* max_msg is used only by the send/receive operations */
+	z->max_msg = (1024 * 1024) - sizeof(struct sock_msg_sendrecv);
```

Comment:
Whatever we set this too, i.e. 1M - sizeof(struct sock_msg_sendrecv), the sockbuf size should be set to 1M.

Reply:
I just pushed more changes. With the changes, the send & receive sockbuf sizes are set to 1M in connect and when handling a connect request.